lib/bluetooth/ble_adv/ble_adv.c

```diff
@@ -72,17 +72,17 @@ static void on_connected(struct ble_adv *ble_adv, ble_evt_t const *ble_evt)
 
 static void on_disconnected(struct ble_adv *ble_adv, ble_evt_t const *ble_evt)
 {
-	int err;
+	uint32_t nrf_err;
 	struct ble_adv_evt adv_evt;
 
 	ble_adv->whitelist_temporarily_disabled = false;
 
 	if (IS_ENABLED(CONFIG_BLE_ADV_RESTART_ON_DISCONNECT)) {
 		if (ble_evt->evt.gap_evt.conn_handle == ble_adv->conn_handle) {
-			err = ble_adv_start(ble_adv, BLE_ADV_MODE_DIRECTED_HIGH_DUTY);
-			if (err) {
+			nrf_err = ble_adv_start(ble_adv, BLE_ADV_MODE_DIRECTED_HIGH_DUTY);
+			if (nrf_err != NRF_SUCCESS) {
```

Comment:
I don't think we should be checking explictly against `NRF_SUCCESS`, it creates lots of noise for nothing.